Formula/iqtree2.rb

```diff
@@ -0,0 +1,31 @@
+class Iqtree2 < Formula
+  desc "Efficient phylogenomic software by maximum likelihood"
+  homepage "http://www.iqtree.org/"
+  url "https://github.com/iqtree/iqtree2.git",
+      tag:      "v2.2.2.5",
+      revision: "ed050c36b1a4cacf220cb47ae6b40df6f2cf15c5"
+  license "GPL-2.0-only"
+
+  depends_on "boost" => :build
+  depends_on "cmake" => :build
+  depends_on "eigen" => :build
+  depends_on "gsl"   => :build
+  depends_on "libomp" => :build
+  depends_on "llvm" => :build
+  depends_on "open-mpi" => :build
+  uses_from_macos "zlib"
+
+  def install
+    inreplace "CMakeLists.txt", "--target=arm64-apple-macos10.5", "-mmacosx-version-min=10.7"
+    inreplace "CMakeLists.txt", "--target=arm64-apple-macos12.0.1", "-mmacosx-version-min=12.0.1"
+    mkdir "build" do
+      system "cmake", "..", "-DIQTREE_FLAGS=omp", "-DCMAKE_C_COMPILER=clang", "-DCMAKE_CXX_COMPILER=clang++"
+      system "make", "-j"
+      bin.install "iqtree2"
+    end
+  end
+
+  test do
+    system "#{bin}/iqtree2", "-v"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.